install_config/monitoring/alerting-rules.adoc

```diff
@@ -0,0 +1,71 @@
+[[alerting-rules]]
+= Alerting rules
+:data-uri:
+:icons:
+:experimental:
+:prewrap!:
+
+OpenShift Cluster Monitoring ships with the following alerting rules configured by default. Currently you cannot add custom alerting rules.
+
+Some alerting rules have identical names. This is intentional. They are alerting on the same event with different thresholds, with different severity, or both. With the inhibition rules, the lower severity is inhibited when the higher severity is firing. This is a best practice.
+
+For more details on the alerting rules, see the link:https://github.com/openshift/cluster-monitoring-operator/blob/master/assets/prometheus-k8s/rules.yaml[configuration file].
```

Comment:
The following alerts are missed, please refer to https://github.com/openshift/cluster-monitoring-operator/blob/master/assets/prometheus-k8s/rules.yaml, and the alert name should followed by alphabetical order. eg, put all KubeCPUOvercommit before KubeMemOvercommit
AlertmanagerDownOrMissing
AlertmanagerFailedReload
ClusterMonitoringOperatorErrors
DeadMansSwitch
EtcdInsufficientMembers
KubeStatefulSetUpdateNotRolledOut
NodeDiskRunningFull (Severity:warning)
PrometheusErrorSendingAlerts (Severity:warning)
PrometheusErrorSendingAlerts (Severity:critical)
PrometheusNotConnectedToAlertmanagers
PrometheusNotIngestingSamples
PrometheusNotificationQueueRunningFull
PrometheusOperatorAlertmanagerReconcileErrors
PrometheusOperatorNodeLookupErrors
PrometheusOperatorPrometheusReconcileErrors
PrometheusTSDBCompactionsFailing
PrometheusTSDBReloadsFailing
PrometheusTSDBWALCorruptions
PrometheusTargetScrapesDuplicate
TargetDown
TelemeterClientDown

Reply:
@juzhao you should probably check with the `release-3.11` branch and not `master` ;) The latter contains stuff for 4.0 already.

Reply:
ok, thanks

Reply:
So the final list of missing alerting rules is the following:

```
AlertmanagerFailedReload
ClusterMonitoringOperatorErrors
DeadMansSwitch
EtcdInsufficientMembers
NodeDiskRunningFull (Severity:warning)
PrometheusErrorSendingAlerts (Severity:warning)
PrometheusErrorSendingAlerts (Severity:critical)
PrometheusNotConnectedToAlertmanagers
PrometheusNotIngestingSamples
PrometheusNotificationQueueRunningFull
PrometheusTSDBCompactionsFailing
PrometheusTSDBReloadsFailing
PrometheusTSDBWALCorruptions
PrometheusTargetScrapesDuplicate
TargetDown
```

@rh-max can you update the documentation, please.

Reply:
A very good catch. Turned out I used a bad macro for converting the .yaml file into the table, so I lost some alerting rules and had wrong descriptions for several other. I re-done the table from scratch. All alerting rules are there, except for AlertmanagerDownOrMissing.